repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,263 @@
+    metadata_url: Optional[str],
+    artifacts_url: Optional[str],
+    artifact_name: str,
+    root: Optional[str],
+    directory_prefix: Optional[str],
+) -> bool:
+    if metadata_url is None:
+        console.print("Please specify metadata url")
+        return False
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if artifacts_url is None:
+        console.print("Please specify artifacts url")
+        return False
+
+    if not os.path.isfile(f"{metadata_dir}/root.json"):
+        console.print(
+            "Trusted local root not found. Using 'tofu' to "
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url} - "
+                "`tofu` was not successful"
+            )
+            return False
+
+    print(f"Using trusted root in {metadata_dir}")
+
+    if directory_prefix:
+        # Annotating as it's covered, but codecov doesn't
+        # recognize that properly
+        download_dir = directory_prefix  # pragma: no cover
+    else:
+        download_dir = os.getcwd() + "/downloads"
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+        )
+        updater.refresh()
+
+        # Annotating, as it's external method we don't want to test
+        info = updater.get_targetinfo(artifact_name)  # pragma: no cover
+
+        if info is None:  # pragma: no cover
+            console.print(f"Artifact {artifact_name} not found")
+            return False
+
+        path = updater.find_cached_target(info)  # pragma: no cover
+        if path:  # pragma: no cover
+            console.print(f"Artifact is available in {path}")
+            return True
+
+        path = updater.download_target(info)  # pragma: no cover
+        console.print(f"Artifact downloaded and available in {path}")
+
+    except (OSError, RepositoryError, DownloadError) as e:
+        console.print(f"Failed to download artifact {artifact_name}: {e}")
+        return False
+
+    return True
+
+
+@artifact.command()
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
```

Comment:
Most likely there is a mistake with this flag as python-tuf already finds the precise bin for an artifact without the need of us to calculate its bin.